tests/client/test_base_client.py

```diff
@@ -469,7 +506,39 @@ async def test_prefect_httpx_client_raises_prefect_http_status_error(
         base_client_send.return_value = RESPONSE_400
         with pytest.raises(PrefectHTTPStatusError) as exc:
             await client.post(url="fake.url/fake/route", data={"evenmorefake": "data"})
-            assert (
-                "Response: {'extra_info': [{'message': 'a test error message'}]}"
-                in str(exc)
-            )
+        expected = "Response: {'extra_info': [{'message': 'a test error message'}]}"
+        assert expected in str(exc)
+
+    @pytest.mark.parametrize(
+        "mock_redirects",
+        [
+            ["https://example.org/redirect_301", "https://example.org/"],
+            ["https://example.org/redirect_302", "https://example.org/redirected_302"],
+            ["https://example.org/redirect_307", "https://example.org/redirected_307"],
+            ["https://example.org/redirect_308", "https://example.org/redirected_308"],
+        ],
+    )
+    async def test_prefect_httpx_client_follow_redirect_true(self, mock_redirects):
+        mock_transport = httpx.MockTransport(redirects)
+        client = PrefectHttpxClient(follow_redirects=True, transport=mock_transport)
+        original_url, redirected_url = mock_redirects
+        response = await client.get(original_url)
+        assert response.url == redirected_url
```

Comment:
This asserts the desired behaviour of the issue the user reported.

Reply:
>  think I'd revise this test a bit and just check that when initializing PrefectClient or CloudClient that follows_redirects is set to True by the code in Prefect and trust that httpx is doing the right thing.

Nods, 💯  Those new tests, are already in:
- `test_prefect_client_follow_redirects`
- `test_cloud_client_follow_redirects`

I worried an integration test was desired and since I had trouble coming up with an elegant way to mock `transport` in the prefect/cloud clients (but could easily for `PrefectHttpxClient`), I wrote tests that asserted the contract for `PrefectHttpxClient` and then tests that asserted the contract for the `PrefectClient` + `CloudClient`, which like lego blocks asserts that they should play nice together.

These `PrefectHttpxClient` tests helped me reproduce the issue and gain confidence in the proposed solution. That said, if I were more familiar with this codebase I would have just gone with these 2 new simple tests, but I wasn't sure that would fly with you all :)
- `test_prefect_client_follow_redirects`
- `test_cloud_client_follow_redirects`

I'll revert these `PrefectHttpxClient` tests.

Thanks SO much for the prompt review!!

Reply:
I totally understand, it's a fuzzy line sometimes on what's an integration test and what's a unit test and guessing what any given repository wants is always hard.